life_expectancy/cleaning.py

```diff
@@ -1,133 +1,110 @@
+    return: clean_dataframe -> clean data
     """
-    #unpivoted_dataframe = pd.melt(raw_dataframe, id_vars='year')
-
 
+    #Unpivot dataframe
     unpivoted_dataframe = pd.melt(raw_dataframe,
-                                id_vars=['unit','sex','age',"geo"],
-                                var_name='year',
-                                value_name = 'value')
-
-
-    return unpivoted_dataframe
-
-def set_and_check_data_types(
-    unpivoted_dataframe: DataFrame
-):
-    """
-    This function sets each columns datatypes according to the assignments specs and scheck them
-
-    param: unpivoted_dataframe -> pandas DataFrame after unpivotting date column
-
-    return: pandas dataframe with columns in the correct data type
-    """
-
-    data_types_dataframe = unpivoted_dataframe
+                                  id_vars = ['unit','sex','age',"geo"],
+                                  var_name = 'year',
+                                  value_name = 'value')
 
+    #Set and check data types
+    dataframe_correct_dtypes = unpivoted_dataframe
     #Cast year column as integer
-    data_types_dataframe['year'] = data_types_dataframe['year'].astype('int')
-
-    #Cast value column as float (deal with null values)
-    data_types_dataframe['value'] = data_types_dataframe['value'].str.split(' ').str[0]
-
-    data_types_dataframe['value'] = data_types_dataframe['value'].replace({':': 0, ': ': 0})
-    data_types_dataframe['value'] = data_types_dataframe['value'].astype('float')
-    data_types_dataframe = data_types_dataframe[data_types_dataframe['value'] != 0]
-
-    data_types_dataframe.columns = ['unit', 'sex', 'age', 'region', 'year', 'value']
-
-    return data_types_dataframe
-
-def filter_by_region(
-    all_regions_dataframe: DataFrame,
-    region: str
-):
-    """
-    This function filters the provided dataframe by the specified region
-
-    :param all_regions_dataframe -> dataframe with every region's data
-    :param region -> region to filter the data by
-
-    return region_filtered_dataframe -> dataframe filtered by region
-    """
-
-    region_filtered_dataframe = all_regions_dataframe[all_regions_dataframe['region'] == region]
-
-    return region_filtered_dataframe
+    dataframe_correct_dtypes['year'] = dataframe_correct_dtypes['year'].astype('int')
+    #Cast value column as float
+    dataframe_correct_dtypes['value'] = dataframe_correct_dtypes['value'].str.split(' ').str[0]
+    #Deal with null values
+    dataframe_correct_dtypes['value'] = dataframe_correct_dtypes['value'].replace({':': -1,
+                                                                                   ': ': -1})
+    dataframe_correct_dtypes['value'] = dataframe_correct_dtypes['value'].astype('float')
+    dataframe_correct_dtypes = dataframe_correct_dtypes[dataframe_correct_dtypes['value'] != -1]
+    #Rename columns
+    dataframe_correct_dtypes.columns = ['unit', 'sex', 'age', 'region', 'year', 'value']
```

Comment:
You should prefer chaining programming in pandas. Everytime you make a change in a column of a dataframe, since dataframes are immutable it give you back a copy of the dataframe.

Reply:
An example of the above is changing the data types of the columns for more than one at once